### PR TITLE
MRG: fix deprecation warnings for sklearn

### PR DESCRIPTION
**Pull request by jasmainak**
closes https://github.com/autoreject/autoreject/issues/65

### CODE REVIEW

File: autoreject/autoreject.py [hunk 1]
```diff
@@ -892,7 +892,6 @@ def fit(self, epochs):
                     local_reject.mean_ = _slicemean(
                         epochs_interp[train].get_data()[:, self.picks],
                         good_epochs_idx, axis=0)
-                    X = epochs[test].get_data()[:, self.picks]
```
Comment (agramfort) [inline autoreject/autoreject.py]: Really ? I are still scoring on the test samples?

Comment (jasmainak) [reply autoreject/autoreject.py]: I am not sure what you mean

Comment (agramfort) [reply autoreject/autoreject.py]: now X is `X = epochs.get_data()[:, self.picks]` while before it was `X = epochs[test].get_data()[:, self.picks]` so it seems score is now called on full epochs not just the test epochs

Comment (agramfort) [reply autoreject/autoreject.py]: forget it. I just saw the X[test] below...

**Comment by agramfort** [comment]
you just need to make CIs happy now :)

File: autoreject/autoreject.py (hunk 1 above)
Comment (jasmainak) [reply autoreject/autoreject.py]: I updated it after seeing your comment :) But just wanted to say there is no bug on master

**Comment by agramfort** [comment]
glad I can still recognize a bug when I see one !

File: autoreject/autoreject.py [hunk 2]
```diff
@@ -869,9 +868,26 @@ def fit(self, epochs):
...
                 verbose=self.verbose)
 
-            for fold, (train, test) in enumerate(_pbar(self.cv, desc='Fold',
-                                                 position=3,
-                                                 verbose=self.verbose)):
+            X = epochs.get_data()[:, self.picks]
+
+            # Hack to allow len(self.cv.split(X)) as ProgressBar
+            # assumes an iterable whereas self.cv.split(X) is a
+            # generator
```
Comment (jasmainak) [inline autoreject/autoreject.py]: @agramfort I had to do this hack. Do you have any better suggestion? Otherwise, this PR is good to go

File: autoreject/autoreject.py [hunk 3]
```diff
@@ -377,15 +376,15 @@ def compute_thresholds(epochs, method='bayesian_optimization',
...
+        cv = StratifiedShuffleSplit(n_splits=10, test_size=0.2,
                                     random_state=random_state)
 
         ch_names = epochs.ch_names
 
         my_thresh = delayed(_compute_thresh)
         verbose = 51 if verbose is not False else 0  # send output to stdout
         threshes = Parallel(n_jobs=n_jobs, verbose=verbose)(
-            my_thresh(data[:, pick], cv=cv, method=method,
+            my_thresh(data[:, pick], cv=cv, method=method, y=y,
```
Comment (agramfort) [inline autoreject/autoreject.py]: I don't understand why you need to pass y around. For me it's always None no?

Comment (jasmainak) [reply autoreject/autoreject.py]: No, it's not always `None`. When we have augmented data, we want to do stratified cross validation so that there are roughly equal number of original trials and augmented trials in each fold

Comment (agramfort) [reply autoreject/autoreject.py]: hum but you use KFold by default which is not stratified. You should use : StratifiedKFold

also it seems you want to use something closer to the groups parameter so you can use GroupKFold. It will allow you to assert that an augmented trial is not part of the test set if the orig trial is in the train set.

Comment (jasmainak) [reply autoreject/autoreject.py]: we do use stratified shuffle split here: https://github.com/autoreject/autoreject/blob/9b05a796eb5e80649de1e988b104a99d53bddd4c/autoreject/autoreject.py#L380

However, it is used only for estimating thresholds, not for estimating `rho` and `kappa`. It is an old open PR but remains to be tested: https://github.com/autoreject/autoreject/pull/28/files.

The goal of this PR is to keep autoreject the same but remove sklearn deprecations

**Comment by agramfort** [comment]
Ok let's move on

Open an issue so we don't forget. 

Now next PR

Go go mainak !
